### PR TITLE
Integrate new FundManager

### PR DESCRIPTION
**Pull request by dirkmc**
# Goals

Integrate with the new Fund Manager in Lotus

# Implementation

- Remove the internal fund manager in go-fil-markets
- convert EnsureFunds to ReserveFunds that both reserves funds and insures they are available in the SMA as needed
- add ReleaseFunds method
- change names of states and events (don't worry, the indexing will not change, they just have new names)
- while the PR may appear long, the only substantive changes are in clientstates.go & providerstates.go -- everything else is just renaming.

### CODE REVIEW

File: storagemarket/impl/clientstates/client_states.go [hunk 1]
```diff
@@ -314,6 +279,17 @@ func FailDeal(ctx fsm.Context, environment ClientDealEnvironment, deal storagema
 	return ctx.Trigger(storagemarket.ClientEventFailed)
 }
 
+func releaseReservedFunds(ctx fsm.Context, environment ClientDealEnvironment, deal storagemarket.ClientDeal) {
+	if !deal.FundsReserved.Nil() && !deal.FundsReserved.IsZero() {
+		err := environment.Node().ReleaseFunds(ctx.Context(), deal.Proposal.Client, deal.FundsReserved)
```
Comment (hannahhoward) [inline storagemarket/impl/clientstates/client_states.go:284]: If you want to use continue using deal.FundsReserved, you probably need to to undelete line 62 above

File: storagemarket/impl/providerstates/provider_states.go [hunk 2]
```diff
@@ -520,14 +499,18 @@ func FailDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal storage
...
 	}
+	releaseReservedFunds(ctx, environment, deal)
+
+	return ctx.Trigger(storagemarket.ProviderEventFailed)
+}
+
+func releaseReservedFunds(ctx fsm.Context, environment ProviderDealEnvironment, deal storagemarket.MinerDeal) {
 	if !deal.FundsReserved.Nil() && !deal.FundsReserved.IsZero() {
-		_, err := environment.DealFunds().Release(deal.FundsReserved)
+		err := environment.Node().ReleaseFunds(ctx.Context(), deal.Proposal.Provider, deal.FundsReserved)
```
Comment (hannahhoward) [inline storagemarket/impl/providerstates/provider_states.go:509]: see same comment from client

**Review by arajasek — APPROVED** [review]
Seems good to me

**Review by dirkmc — COMMENTED** [review]
LGTM 👍

**Pull request merged**